documentation/modules/security/ref-certificates-and-secrets.adoc

```diff
@@ -0,0 +1,151 @@
+// Module included in the following assemblies:
+//
+// assembly-security.adoc
+
+[id='certificates-and-secrets-{context}']
+= Secrets
+
+Strimzi uses _Secrets_ to store private keys and certificates for Kafka cluster components and clients.
+Secrets are used for establishing TLS encrypted connections between Kafka brokers, and between brokers and clients.
+They are also used for mutual TLS authentication.
+
+* A _Cluster Secret_ contains a cluster CA certificate to sign Kafka broker certificates, and is used by a connecting client to establish a TLS encrypted connection with the Kafka cluster to validate broker identity.
+* A _Client Secret_ contains a client CA certificate for a user to sign its own client certificate to allow mutual authentication against the Kafka cluster.
+The broker validates the client identity through the client CA certificate itself.
+* A _User Secret_ contains a private key and certificate, which are generated and signed by the client CA certificate when a new user is created.
+The key and certificate are used for authentication and authorization when accessing the cluster.
+
+Secrets provide private keys and certificates in PEM and PKCS #12 formats.
+Using private keys and certificates in PEM format means that users have to get them from the Secrets, and generate a corresponding truststore (or keystore) to use in their Java applications.
+PKCS #12 storage provides a truststore (or keystore) that can be used directly.
+
+All keys are 2048 bits in size.
+
+CA certificate validity periods are expressed as a number of days after certificate generation.
+You can configure the validity period of:
+
+* Cluster CA certificates in `Kafka.spec.clusterCa.validityDays`
+* Client CA certificates in `Kafka.spec.clientsCa.validityDays`
+
+== PKCS #12 storage
```

Comment:
I think this chapter needs some restructuring. The main chapter is called _Secrets_ but starts talking about CA certificates and their validity which has nothing to do with Secrets. Then it has this as a subchapter, which suggests it is related to the previous, but it is not really related to the CAs, the main CA secrets don't even use PKCS12. So this seems a bit confusing.

Reply:
The main chapter heading is actually 'Security', then we introduce the concepts of CA authorities and Secrets.  I'll see if we can restructure to avoid any potential confusion and group related content together.

Reply:
I moved the validity section under CA certificates. I think it makes better sense now. I also think the PKCS 12 section is now fine here too. We talk about secrets formats before that sub-section and then we follow with a description of Secret with info on formats.